pkg/ring/kv/memberlist/memberlist_client.go

```diff
@@ -433,6 +433,7 @@ func (m *Client) CAS(ctx context.Context, key string, f func(in interface{}) (ou
 	}
 
 	sleep := false
+	failReason := "too many retries"
```

Comment:
May you enrich it saying how many retries.

Reply:
there is hardcoded 10 retries in the code :) let's not point that fact out too much.